File: app/src/main/java/com/lemonade/dima/di/ApplicationComponent.java
package com.lemonade.dima.di;

import com.lemonade.dima.App;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.AndroidInjector;

/*@Singleton*/
@SuppressWarnings("WeakerAccess")
@Component(modules = {/*AndroidInjectionModule.class*//*, AppModule.class*/})
public interface ApplicationComponent /*extends AndroidInjector<App>*/ {
    MessagesActivityComponent plusMessagesActivityComponent(MessagesActivityModule module);
}

File: app/src/main/java/com/k4dima/chat/features/app/ApplicationComponent.kt
package com.k4dima.chat.features.app

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [AndroidSupportInjectionModule::class, AppBindingModule::class])
internal interface ApplicationComponent : AndroidInjector<App>